Marlin/src/sd/cardreader.cpp

```diff
@@ -425,7 +425,9 @@ void CardReader::manage_media() {
 
     if (stat) {                       // Media Inserted
       safe_delay(500);                // Some boards need a delay to get settled
-      mount();                        // Try to mount the media
+      if (ENABLED(SD_MOUNT_ON_INSERT) && old_stat == 2) {
+        mount();                      // Try to mount the media
+      }                            
```

Comment:
This change breaks Marlin's SD card handling.

- Before: Mount the SD card if an insert is detected
- `SD_MOUNT_ON_INSERT` enabled: Only mount SD here when first booting up
- `SD_MOUNT_ON_INSERT` disabled: Never mount an inserted SD card

I'll see if I can fix this so it matches your stated intention.

Reply:
This would be nice, i´d also take an explanation then if you got some time (maybe on discord) :) 
Or maybe i can help you with this.